src/runtime/WeakMapObject.cpp

```diff
@@ -74,6 +74,35 @@ Value WeakMapObject::get(ExecutionState& state, PointerValue* key)
     return Value();
 }
 
+Value WeakMapObject::getOrInsert(ExecutionState& state, PointerValue* key, const Value& value)
+{
+    ASSERT(key->isObject() || key->isSymbol());
+    for (size_t i = 0; i < m_storage.size(); i++) {
+        auto existingKey = m_storage[i]->key.unwrap();
+        if (existingKey && existingKey == key) {
+            return m_storage[i]->data;
+        }
+    }
+
+    for (size_t i = 0; i < m_storage.size(); i++) {
+        if (!m_storage[i]->key) {
+            m_storage[i]->key = key;
+            m_storage[i]->data = value;
+            GC_GENERAL_REGISTER_DISAPPEARING_LINK_SAFE(reinterpret_cast<void**>(&m_storage[i]->key), key);
```

Comment:
Memory leak of `data` has been resolved in #1464 
please add the following statement
```c++
GC_GENERAL_REGISTER_DISAPPEARING_LINK_SAFE(reinterpret_cast<void**>(&m_storage[i]->data), key);
```